src/stream/src/executor/over_window/state/lag.rs

```diff
@@ -16,24 +16,24 @@ use std::collections::VecDeque;
 
 use risingwave_common::must_match;
 use risingwave_common::types::Datum;
-use risingwave_expr::function::window::Frame;
+use risingwave_expr::function::window::{Frame, FrameBound};
 use smallvec::SmallVec;
 
 use super::{StateKey, StateOutput, StatePos, WindowState};
 use crate::executor::over_window::state::StateEvictHint;
+use crate::executor::StreamExecutorResult;
 
 struct BufferEntry(StateKey, Datum);
 
 pub(super) struct LagState {
     offset: usize,
-    // TODO(rc): may move buffer maintenance to a common structure.
     buffer: VecDeque<BufferEntry>,
     curr_idx: usize,
 }
 
 impl LagState {
     pub fn new(frame: &Frame) -> Self {
-        let offset = must_match!(frame, Frame::Offset(offset) if *offset < 0 => -offset as usize);
+        let offset = must_match!(frame, Frame::Rows(FrameBound::Preceding(offset), FrameBound::CurrentRow) => *offset);
```

Comment:
What's the difference between our `must_match` and `assert_matches`?

Reply:
Looks the same😄 Will change to `assert_matches`.